### PR TITLE
change default temperature of OAI compat API from 0 to 1

### PR DESCRIPTION
**Pull request by Kartoffelsaft**
This should make the API more similar to that of OpenAI's actual API

### CODE REVIEW

**Review by mofosyne — APPROVED** [review]
Double checking your assertion, can confirm that at least for chat completion mode, which is what we are dealing with this PR. The default is indeed temperature=1.0

Source: https://platform.openai.com/docs/api-reference/chat/create#chat-create-temperature

> temperature
> number or null
> The sampling temperature used for this run. If not set, defaults to 1.

Just a quick note that this is example code not the actual llama.cpp endpoint itself. But still be useful to maintain consistency.

---

Note that when in transcript mode, creativity/temperature is by default 0. So temperature defaults can differ between different api endpoints.

**Comment by shibe2** [comment]
Different models can tolerate different temperatures. What if 1 is too high for most models that people run locally? Default in main is 0.8.

**Comment by jukofyork** [comment]
> Different models can tolerate different temperatures. What if 1 is too high for most models that people run locally? Default in main is 0.8.

The value of 1 should work for any model assuming the logits weren't scaled whilst training.

The value of 1 actually corresponds to the model outputting "well calibrated" probability estimates, ie: if you were to plot the post-softmax probability estimates along with the empirical fraction of times the next token fell in the respective "bin" (or the log of these values more likely), then assuming log-loss (aka "cross entropy" loss) was used; you'd find that temperature=1 would make the plots line up the best.

The inverse of this is even used to calibrate non-probabilistic models' outputs (eg: SVM using "maximum margin" loss):

https://en.m.wikipedia.org/wiki/Platt_scaling
